diff --git a/src/pyadjoint/adjoint_source_types/multitaper_misfit.py b/src/pyadjoint/adjoint_source_types/multitaper_misfit.py
index 5f0b8bd..1a029c3 100644
--- a/src/pyadjoint/adjoint_source_types/multitaper_misfit.py
+++ b/src/pyadjoint/adjoint_source_types/multitaper_misfit.py
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-# -*- encoding: utf8 -*-
+# -*- encoding: utf-8 -*-
 """
 Multitaper based phase and amplitude misfit and adjoint source.
 
@@ -781,6 +781,15 @@ def calculate_adjoint_source(observed, synthetic, config, window,
         cc_tshift = cc_shift * deltat
         cc_dlna = 0.5 * np.log(sum(d**2) / sum(s**2))
 
+        import obspy
+        t = obspy.core.trace.Trace(d, {'dt': 1.0})
+        st = obspy.core.stream.Stream(traces=[t])
+        st.write("data.sac", "SAC")
+
+        t = obspy.core.trace.Trace(s, {'dt': 1.0})
+        st = obspy.core.stream.Stream(traces=[t])
+        st.write("syn.sac", "SAC")
+
         # uncertainty estimate based on cross-correlations
         sigma_dt_cc = 1.0
         sigma_dlna_cc = 1.0
@@ -809,6 +818,12 @@ def calculate_adjoint_source(observed, synthetic, config, window,
         else:
             raise Exception
 
+
+        t = obspy.core.trace.Trace(d, {'dt': 1.0})
+        st = obspy.core.stream.Stream(traces=[t])
+        st.write("data.recon.sac", "SAC")
+
+
         # ===
         # Make decision wihich method to use: c.c. or multi-taper
         # always starts from multi-taper, if it doesn't work then
